packages/react/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -373,14 +373,29 @@ const MultiSelect = React.forwardRef(
       selectedItems: selected,
     });
 
+    // Filter out items with an object having undefined values
+    const filteredItems = items.filter((item) => {
+      if (typeof item === 'object' && item !== null) {
+        for (const key in item) {
+          if (
+            Object.prototype.hasOwnProperty.call(item, key) &&
+            item[key] === undefined
+          ) {
+            return false; // Return false if any property has an undefined value
+          }
+        }
+      }
+      return true; // Return true if item is not an object with undefined values
+    });
+
```

Comment:
I think it would make sense for this to be in a `useMemo` with `items` in the dependency array so it doesn't filter on every render.